Tests/ComposableArchitectureTests/ReducerTests.swift

```diff
@@ -97,16 +171,72 @@ final class ReducerTests: BaseTCATestCase {
     var first = false
     var second = false
 
+    let _store = Store(
+      initialState: State(counter: 0),
+      reducer: CombineReducers {
+        One(effect: { @MainActor in first = true })
+        Two(effect: { @MainActor in second = true })
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }.onChange(of: \.counter) { _, _, _ in
+          .send(.increment)
+      }
```

Comment:
Now the sheer amount of chaining in this example does not compile even when hoisted into a `body`, and the problems is exponential:

- 9 chains: 3 seconds
- 10 chains: 6 seconds
- 11 chains: 12 seconds
- 12 chains: 30 seconds

If these compile quickly in Xcode 14.2, you may have stumbled on a big type checking regression in Swift 5.8, so it might be good to file an issue with Apple.